qa/setup_packages.py

```diff
@@ -105,6 +106,22 @@ def cal_num_of_configs(use, cuda):
         ret *= len(get_package(packages, key, cuda))
     return ret
 
+def get_all_string(use, cuda):
+    ret = []
+    for key in packages.keys():
+        if key not in use:
+            continue
+        for val in get_package(packages, key, cuda):
+            if val is None:
+                ret.append(key)
+            elif 'http' in val:
```

Comment:
What is that supposed to do? Won't it be triggered for any val that happens to contain `http` as a substring?

Reply:
It is copied from `get_install_string`. It differs versions given as urls from those expressed with numbers. It actually needs to handle only those listed in `packages` in this script and it works for them.

Reply:
```suggestion
            elif val.startswith('http'):
```
`get_install_string` could be adjusted as well